api/client.go

```diff
@@ -507,23 +520,29 @@ func (c *Client) getSinkPathFromAgent(agentAddress string) (string, error) {
 }
 
 // starts a go routine to poll the specified file for a token
-func (c *Client) pollFileForToken(filePath string) {
+func (c *Client) pollFileForToken(filePath string, quit chan int) {
 	pollingInterval := 60 * time.Second //todo - what should this be?
 	go func() {
 		for {
-			time.Sleep(pollingInterval)
-			token, err := readAgentTokenFromFile(filePath) // todo - what to do with the error?
-			// update the client's token if it has changed and there was no error reading the file
-			if err == nil && token != c.token {
-				c.modifyLock.Lock()
-				c.token = token
-				c.modifyLock.Unlock()
+			select {
+			case <-quit:
+				quit <- 1
+				return
+			default:
+				time.Sleep(pollingInterval)
+				token, err := readTokenFromFile(filePath) // todo - what to do with the error?
```

Comment:
Maybe instead of messing with channel closes and such, simplest thing would be to always start the polling function, always have it running, but add an `if` here that does nothing if the tokenFilePath is empty.

Reply:
The channel is never actually closed, it gets re-used when it is needed again.

I like putting the `if` logic into the polling routine and just letting it run. Opting for that as It will cleanup some of the other comments as well (writing to quit and read/write instead of closing c.pollingChannel)